src/main/java/codesquad/springcafe/CheckLoginInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package codesquad.springcafe;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
+import jakarta.websocket.Session;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+public class CheckLoginInterceptor implements HandlerInterceptor {
+
+    private static final Logger logger = LoggerFactory.getLogger(CheckLoginInterceptor.class);
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        return true;
+    }
+
+    @Override
+    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler, ModelAndView modelAndView) throws Exception {
+        HttpSession session = request.getSession();
+        String viewName = null;
+        if(modelAndView != null) {
+            viewName = modelAndView.getViewName();
+        }
+        // 리다이렉트의 경우 ModelAndView가 AddObject된 요소들을 쿼리 파라미터에 담아 전달하기 때문에 조건문을 달았습니다.
+        // Location : localhost:8080/;?loginUserId=dao
+        if (viewName != null && !viewName.startsWith("redirect:")) {
+            if (session != null) {
+                Object loginUserId = session.getAttribute("loginUserId");
```

Comment:
Objet가 아니라 캐스팅을 해서 쓰는 게 좋습니다.
```suggestion
                String loginUserId = (String) session.getAttribute("loginUserId");
```